### DISCUSSION TITLE
Welcome to evm-dafny Discussions!

### DISCUSSION THREAD

**Discussion by DavePearce** [ConsenSys/evm-dafny#580] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by gaxiiiiiiiiiiii** [comment]
Hello everyone,

I have been writing Coq as a hobby and have a keen interest in Crypto. I had the opportunity to contribute, albeit briefly, to a project that involved verifying the Tezos Blockchain using Coq. In that project, I wrote [a few proofs for basic functions](https://gitlab.com/formal-land/coq-tezos-of-ocaml/-/merge_requests?scope=all&state=merged&author_username=gaxiiiiiiiiiiii). 

I've always thought that formal verification in Ethereum would be exciting, which is why I find the evm-dafny project remarkable! However, I have no experience in smart contract development and only started learning Dafny last month. So, I plan to learn both Dafny and EVM through my exploration of evm-dafny.

My experience with Coq, especially in writing simple mathematical proofs, could be beneficial. For instance, I have recently made a [pull request related to the Pow function](https://github.com/Consensys/DafnyCrypto/pull/20).

Looking forward to contributing and learning with you all. Thank you.

**Reply by DavePearce** [reply]
Hey @gaxiiiiiiiiiiii, welcome to DafnyEVM!  We are certainly keen to see formal verification of smart contracts become a routine thing, and the DafnyEVM is one step in this direction.  However, there remains a lot to be done in terms of making an easy-to-use system.  We are doing our best to figure out strategies and approaches here.  You can see some example proofs using the DafnyEVM [here](https://github.com/DavePearce/BettingContract/tree/main/proof) and [here](https://github.com/Consensys/WrappedEther.dfy/tree/main/src).
